### PR TITLE
Add --thin option to devkit

### PR DESCRIPTION
**Pull request by JohnCoates**
Used for macOS and iOS thin builds.

Modifies script to use argparse module. Not sure what the correct terminology for the type of attribute "thin" is, I termed it "variant".

### CODE REVIEW

**Comment by oleavr** [comment]
Sweet!

"Variant" is not bad, but I'd prefer to go for something else since it's already used in the context of [GLib.Variant](https://developer.gnome.org/glib/stable/glib-GVariant.html).  How about naming it `flavor` instead, and describing the option as `build without cross-arch support`? (To match the Makefile help output.)

**Comment by JohnCoates** [comment]
Sounds good. Just pushed the suggestions

File: releng/devkit.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 import subprocess
 import sys
 import tempfile
+import argparse
```
Comment (oleavr) [inline releng/devkit.py]: Would be good to move this right before `import codecs`, so the imports are kept in alphabetical order.

File: releng/devkit.py [hunk 2]
```diff
@@ -528,17 +529,26 @@ def deduplicate(items):
...
-
-    kit = sys.argv[1]
-    host = sys.argv[2]
-    outdir = os.path.abspath(sys.argv[3])
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument("kit")
+    parser.add_argument("host")
+    parser.add_argument("outdir")
+    parser.add_argument("-t", "--thin", help='build without cross-arch support', action="store_true")
```
Comment (oleavr) [inline releng/devkit.py]:
```suggestion
    parser.add_argument("-t", "--thin", help="build without cross-arch support", action='store_true')
```

(Should use double quotes for regular strings, and single quotes for enum/symbol values. In hindsight this is a bit of a weird convention, but better stick to it for consistency and consider changing it globally later.)

File: releng/devkit.py (hunk 1 above)
Comment (oleavr) [reply releng/devkit.py]: Actually this import is already below so we should drop it here.

File: releng/devkit.py [hunk 3]
```diff
@@ -528,17 +529,26 @@ def deduplicate(items):
...
 
 if __name__ == "__main__":
-    if len(sys.argv) != 4:
-        print("Usage: {0} kit host outdir".format(sys.argv[0]), file=sys.stderr)
-        sys.exit(1)
-
-    kit = sys.argv[1]
-    host = sys.argv[2]
-    outdir = os.path.abspath(sys.argv[3])
+    import argparse
```
Comment (oleavr) [inline releng/devkit.py]:
```suggestion
    import argparse

```

File: releng/devkit.py (hunk 2 above)
Comment (JohnCoates) [reply releng/devkit.py]: Oops, good catch!

File: releng/devkit.py (hunk 1 above)
Comment (JohnCoates) [inline releng/devkit.py]:
```suggestion

```

**Comment by oleavr** [comment]
Thanks!

**Comment by JohnCoates** [comment]
You're welcome. Glad to help

**Pull request merged**